lib/rouge/lexer.rb

```diff
@@ -501,12 +521,47 @@ def self.detect?(text)
   end
 
   module Lexers
+    ProxyLexer = Struct.new :path, :name, :tag, :aliases, :filenames, :mimetypes
+
+    @_loaded_files = {}
     @_loaded_lexers = {}
+    @_loaded_proxies = {}
+    @_source_files = []
+
+    def self.add_lexer(class_name)
+      return if @_source_files.empty?
+      @_loaded_files[class_name] = @_source_files.last
+    end
 
     def self.load_lexer(relpath)
       return if @_loaded_lexers.key?(relpath)
       @_loaded_lexers[relpath] = true
+      @_source_files.push relpath
       load File.join(__dir__, 'lexers', relpath)
+      @_source_files.pop
+    end
+
+    def self.load_proxies(hashes)
+      hashes.each do |h|
+        next if h["tag"].nil?
+        proxy = ProxyLexer.new(h["path"], h["name"].to_sym, h["tag"],
+                               h["filenames"], h["mimetypes"])
+        @_loaded_proxies[proxy.name] = proxy
+      end
+
+      registry = {}
+      @_loaded_proxies.values.each do |proxy|
```

Comment:
Prefer iterating directly:
```suggestion
      @_loaded_proxies.each_value do |proxy|
```

Reply:
I simplified this code and got rid of this part.